tests/unit/src/main/scala/tests/BuildServerInitializer.scala

```diff
@@ -0,0 +1,110 @@
+package tests
+
+import scala.concurrent.Future
+import scala.util.Properties
+
+import scala.meta.internal.builds.BuildTool
+import scala.meta.internal.metals.Debug
+import scala.meta.internal.metals.Messages.ImportBuild
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.metals.ServerCommands
+
+import ch.epfl.scala.bsp4j.BspConnectionDetails
+import com.google.gson.Gson
+import org.eclipse.lsp4j.MessageActionItem
+
+trait BuildServerInitializer {
+  def initialize(layout: String, expectError: Boolean = false): Future[Unit]
+}
+
+trait QuickBuildInitializer extends BuildServerInitializer {
+  this: BaseLspSuite =>
+  override def initialize(
+      layout: String,
+      expectError: Boolean = false
+  ): Future[Unit] = {
+    Debug.printEnclosing()
+    writeLayout(layout)
+    QuickBuild.bloopInstall(workspace)
+    for {
+      _ <- server.initialize()
+      _ <- server.initialized()
+    } yield {
+      if (!expectError) {
+        server.assertBuildServerConnection()
+      }
+    }
+  }
+}
+
+trait BloopImportInitializer extends BuildServerInitializer {
+  this: BaseLspSuite =>
+  override def initialize(
+      layout: String,
+      expectError: Boolean = false
+  ): Future[Unit] = {
+    writeLayout(layout)
+    for {
+      _ <- server.initialize()
+      // Import build using Bloop
+      _ = client.importBuild = ImportBuild.yes
+      _ <- server.initialized()
+    } yield {
+      if (!expectError) {
+        server.assertBuildServerConnection()
+      }
+    }
+  }
+}
+
+trait SbtServerInitializer extends BuildServerInitializer {
+  this: BaseLspSuite =>
+  override def initialize(
+      layout: String,
+      expectError: Boolean = false
+  ): Future[Unit] = {
+    writeLayout(layout)
+    generateBspConig()
+    for {
+      _ <- server.initialize()
+      _ <- server.initialized()
+      // choose sbt as the Bsp Server
+      _ = client.selectBspServer = { _ => new MessageActionItem("sbt") }
```

Comment:
Since we just chose sbt here, do we still need the line below this where we do a `bsp-switch` and choose sbt again?

Reply:
This line means that the client will respond with "sbt" if prompted to choose a bsp server. The next line triggers the switch, which sends the `SelectBspServer` message to the client. So both lines are useful.

Yet I think the "sbt" parameter in `server.executeCommand(ServerCommands.BspSwitch.id, "sbt")` is useless because the `BspSwitch` command doesn't expect a parameter:
https://github.com/scalameta/metals/blob/3c83447ec658f87fdccbfb3f0a39fca1cec4ef6e/metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala#L1601

Reply:
Ahhh gotcha gotcha. That make sense 👍🏼